### PR TITLE
ip headers list [EN-3811]

### PR DESCRIPTION
**Pull request by yaronpx**

### CODE REVIEW

File: src/PerimeterxContext.php [hunk 1]
```diff
@@ -68,6 +62,27 @@ public function __construct($pxConfig)
         $this->sensitive_route = $this->checkSensitiveRoutePrefix($pxConfig['sensitive_routes'], $this->uri);
     }
 
+    private function extractIP($pxConfig) {
+        $pxConfig['logger']->warning('looking for ip');
```
Comment (nitzanpx) [inline src/PerimeterxContext.php]: change to info

File: src/PerimeterxContext.php [hunk 2]
```diff
@@ -68,6 +62,27 @@ public function __construct($pxConfig)
...
+        $all_headers = getallheaders();
+
+        if (isSet($pxConfig['ip_headers'])) {
+            foreach ($pxConfig['ip_headers'] as $header) {
+                if (isSet($all_headers[$header])) {
```
Comment (nitzanpx) [inline src/PerimeterxContext.php]: change isset

Comment (yaronpx) [inline src/PerimeterxContext.php]: believe it or not, it works this way too! yay PHP. ..
but I'll change it

File: src/PerimeterxContext.php [hunk 3]
```diff
@@ -68,6 +62,27 @@ public function __construct($pxConfig)
...
+                    return $all_headers[$header];
+                }
+            }
+        }
+        if (isset($pxConfig['custom_user_ip'])) {
+            return $pxConfig['custom_user_ip']($this);
+        }
+        if (function_exists('pxCustomUserIP')) {
```
Comment (nitzanpx) [inline src/PerimeterxContext.php]: `function_exists`, where should this function exist? 
shouldn't this function be extracted from `$pxConfig`?

and also, there is no documentation for this, is it setting the ip variable in the `$ctx`?

**Comment by nitzanpx** [comment]
@yaronpx  tests on 5.6 dont pass

**Pull request merged**